specs/merge/p2p-interface.md

```diff
@@ -0,0 +1,111 @@
+Readers should understand the Phase 0 and Altair documents and use them as a basis to understand the changes outlined in this document.
+
+## Table of contents
+
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+  - [Warning](#warning)
+- [Modifications in the Merge](#modifications-in-the-merge)
+  - [The gossip domain: gossipsub](#the-gossip-domain-gossipsub)
+    - [Topics and messages](#topics-and-messages)
+      - [Global topics](#global-topics)
+        - [`beacon_block`](#beacon_block)
+    - [Transitioning the gossip](#transitioning-the-gossip)
+  - [The Req/Resp domain](#the-reqresp-domain)
+    - [Messages](#messages)
+      - [BeaconBlocksByRange v2](#beaconblocksbyrange-v2)
+      - [BeaconBlocksByRoot v2](#beaconblocksbyroot-v2)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Warning
+
+This document is currently illustrative for early Merge testnets and some parts are subject to change.
+Refer to the note in the [validator guide](./validator.md) for further details.
+
+# Modifications in the Merge
+
+## The gossip domain: gossipsub
+
+Some gossip meshes are upgraded in the Merge to support upgraded types.
+
+### Topics and messages
+
+Topics follow the same specification as in prior upgrades.
+All topics remain stable except the beacon block topic which is updated with the modified type.
+
+The specification around the creation, validation, and dissemination of messages has not changed from the Phase 0 and Altair documents.
+
+The derivation of the `message-id` remains stable.
+
+The new topics along with the type of the `data` field of a gossipsub message are given in this table:
+
+| Name | Message Type |
+| - | - |
+| `beacon_block` | `SignedBeaconBlock` (modified) |
+
+Note that the `ForkDigestValue` path segment of the topic separates the old and the new `beacon_block` topics.
+
+#### Global topics
+
+The Merge changes the type of the global beacon block topic.
+
+##### `beacon_block`
+
+The existing specification for this topic does not change from prior upgrades,
+but the type of the payload does change to the (modified) `SignedBeaconBlock` found in the Merge.
+This type changes due to the addition of `execution_payload` to the inner `BeaconBlockBody`.
+
+See the Merge [state transition document](./beacon-chain.md#beaconblockbody) for further details.
+
+### Transitioning the gossip
+
+See gossip transition details found in the [Altair document](../altair/p2p) for
+details on how to handle transitioning gossip topics for the Merge.
+
+## The Req/Resp domain
+
+### Messages
+
+#### BeaconBlocksByRange v2
+
+**Protocol ID:** `/eth2/beacon_chain/req/beacon_blocks_by_range/2/`
+
+Request and Response remain unchanged.
+`MERGE_FORK_VERSION` is used as an additional `context` to specify the Merge block type.
```

Comment:
```suggestion
The Merge fork-digest is introduced to the `context` enum to specify the Merge block type.
```

Nitpick: it's the fork-digest, not the fork-version, which is used as context. And I think of the context as an `enum`, something we extend, but not have more instances of. It's not version `/3`, just a backwards-compatible change to `/2`

Reply:
right. I kept it as /2 but your phrasing is clearer that it is an extension of the valid values. fixed!